code/modules/photography/camera/camera.dm

```diff
@@ -39,6 +38,13 @@
 	var/can_customise = TRUE
 	var/default_picture_name
 
+	var/see_ghosts = CAMERA_NO_GHOSTS //for the spoop of it
+	var/static/list/detectable_invisible_atom = list(
+		/mob/dead/observer,
+		/mob/living/simple_animal/revenant,
+		/mob/living/simple_animal/hostile/floor_cluwne
+	)
```

Comment:
Use an associative list and use lookups instead of 'in'.
(I think byond's dictionary implementation is good, so going off of regular implementations this should be true)

Using in requires checking every element in the list to see if it is correct. This is O(n) (The time it takes to execute this is proportional to the size of the list).
```
for(var/i in list)
  if (i == thing)
    return i
return null
```

Using a dictionary lookup generates a hash and then uses that to check if the element is in the dictionary. This is O(1) (No matter how many elements are in the list, it takes the same amount of time to check)
```
//The hash is an index of the list in the range 0 to n (n >= number of elements in the list (The more nulls there are in the list, the more memory you waste, but the more optimised this is as you get less hash collisions))
hashed_index = calc_hash(thing)
var/located = list[hashed_index]
//To account for hash collisions (2 things have the same hash)
//Keep looking through the list until we find null, or the thing we are looking for
//If 2 things have the same hash, then it will be inserted in the next free slot
while (located  != null)
  if (located == thing)
    return thing
  hashed_index ++
  located  = list[hashed_index]
//Its not in the list
return null
```

TL;DR dictionaries have better scaling factor, so using a lookup will probably be better here (Although since n is 3 it probably doesn't make a whole lot of difference).

Reply:
```dm
var/static/list/detectable_invisible_atom = list(
/mob/dead/observer = TRUE,
/mob/living/simple_animal/revenant = TRUE,
/mob/living/simple_animal/hostile/floor_cluwne = TRUE
)

if (detectable_invisible_atom[/mob/living/simple_animal/hostile/floor_cluwne])
  //We can see the thing!
  see_thing()
```

Reply:
wow giving TRUE for each key like that. interesting.